src/components/forms/formr-part-a/ValidationSchema.tsx

```diff
@@ -6,120 +6,75 @@ const mobileRegex = /((\+44(\s\(0\)\s|\s0\s|\s)?)|0)7\d{3}(\s)?\d{6}/g;
+
+const ValidationSchema = yup.object({
+  forename: stringSchema("Forename(s)"),
+  surname: stringSchema("Surname (GMC-Registered)", 30),
+  gmcNumber: stringSchema("GMC number", 20),
+  localOfficeName: stringSchema("Deanery / HEE Local Office"),
+  dateOfBirth: dateSchema("Your date of birth").test(
+    "dateOfBirth",
+    "You must be 17 years or above",
+    value => DateUtilities.IsLegalAge(value)
+  ),
+  gender: stringSchema("Gender"),
+  immigrationStatus: stringSchema("Immigration Status"),
+  qualification: stringSchema("Qualification"),
+  dateAttained: dateSchema(
+    "Date awarded (most recent qualification)"
+  ).test(
+    "dateAttained",
+    "Date awarded (most recent qualification) - please choose a date from the past",
+    value => DateUtilities.IsPastDate(value)
+  ),
+  medicalSchool: stringSchema("Medical school", 100),
+  address1: stringSchema("Address - house number/ name and road"),
+  address2: stringSchema("Address - district"),
+  address3: stringSchema("Address - town or city"),
+  address4: stringSchema("Address - country"),
+  postCode: stringSchema("Postcode", 8).matches(
+    postcodeRegex,
+    "Please enter a valid postcode"
+  ),
+  telephoneNumber: stringSchema("Contact (landline telephone)").matches(
+    phoneRegex,
+    "Contact (landline) number - requires a valid number"
+  ),
+  mobileNumber: stringSchema("Contact (mobile)").matches(
+    mobileRegex,
+    "Contact (mobile) number - requires a valid number"
+  ),
+  email: stringSchema("Email").email("Email is invalid"),
   declarationType: yup
     .string()
     .required("You need to choose at least one Declaration")
     .nullable(),
-  programmeSpecialty: yup
-    .string()
-    .max(50)
-    .required("Programme specialty is required"),
-  college: yup
-    .string()
-    .max(50)
-    .required("Royal College / Faculty Assessing Training is required"),
-  completionDate: yup
-    .date()
-    .required("Anticipated completion date is required")
-    .test(
-      "completionDate",
-      "Anticipated completion date - please choose a future date",
-      value => DateUtilities.IsFutureDate(value)
-    ),
-  trainingGrade: yup
-    .string()
-    .max(50)
-    .required("Training Grade is required"),
-  startDate: yup.date().required("Programme start date is required"),
-  programmeMembershipType: yup
-    .string()
-    .max(50)
-    .required("Post type / Appointment is required"),
+  programmeSpecialty: stringSchema("Programme specialty"),
+  college: stringSchema("Royal College / Faculty Assessing Training"),
+  completionDate: dateSchema(
+    "Anticipated completion date"
+  ).test(
+    "completionDate",
+    "Anticipated completion date - please choose a future date",
+    value => DateUtilities.IsFutureDate(value)
+  ),
+  trainingGrade: stringSchema("Training Grade"),
+  startDate: dateSchema("Programme start date"),
+  programmeMembershipType: stringSchema("Post type / Appointment"),
   wholeTimeEquivalent: yup
```

Comment:
Still not sure about the validation for the WTE. For example, the error we get for whitespace is not very user-friendly. A cleaner solution might be a single regex to test the string input - something like this: `/^((0\.[1-9]{1})?|(0\.([0-9]{1}[1-9]{1}|[1-9]{1}[0-9]{1}))|1(\.0{1,2})?)$/` - with just a single error message. Happy to push suggested changes to this branch if agreed. Cheers

Reply:
@john12321 Sure. Please push the change.